scripts/precompile-headers.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+BIN=$(dirname "$0")/
+HILTI_CONFIG=${BIN}/hilti-config
+
+for flag in $(${HILTI_CONFIG} --cxxflags); do
+    if ! echo "${flag}" | grep -q '^-I'; then
+        continue
+    fi
+    dir=${flag#??}
+    if [ -e "${dir}"/hilti/rt/libhilti.h ]; then
+        LIBHILTI=${dir}/hilti/rt/libhilti.h
+    fi
+done
+
+if [ -z "${var+x}" ]; then
+    echo "Error: could not determine location of libhilti.h"
+    exit 1
+fi
+
+VERSION=$(${HILTI_CONFIG} --version | cut -d ' ' -f1)
+
+CACHE=$HOME/.cache/spicy/${VERSION}
```

Comment:
Per above, I think we need a way to relocate the cache dir elsewhere, either through a script option or an environment variable.

Reply:
I went with the environment variable.